### DISCUSSION TITLE
Change livewire default app.layout to custom package layout

### DISCUSSION THREAD

**Discussion by sharifghaderi** [livewire/livewire#4601] [1. Help]
Hi, 
What I'm trying to do is to create a package based on livewire. 

When I try routing like this: 
`
Route::get('/admin', function () {
    return view('mypackage::dashboard');
});
`

everything works fine.

But when I try to call a route with Controller class like this:

`
Route::get('admin/edit-page/{id}',EditPage::class);
`

Where EditPage is a Livewire component, I get the following error:

`
View [layouts.app] not found. (View: C:\xampp\xampp2\htdocs\testproject\vendor\livewire\livewire\src\Macros\livewire-view-component.blade.php)
`
This is somehow clear to me and in the documentation there is a part which explains where we can change this but it still won't look in my package directory but in the default laravel view directory or the app directory.

**Just to mention:**
I want that this routing works with for example this layout:
`/packages/testpackage/testpackage/src/resources/views/layout/master.blade.php`

This is the blade template which all other livewire components are working with but it just fails when I give a class parameter to the my route:

![grafik](https://user-images.githubusercontent.com/14842576/152148799-f02dd086-4529-4a9f-9d74-4431f90d936a.png)

**Comment by joshhanley** [comment]
@sharifghaderi hmm you should be able to pass a custom layout into the `->layout()` method in the render method. You will need to make sure your have registered it and given it a namespace, and then you should be able to do `->layout('mypackage::mylayout')`.

I haven't tested it though, so can you try that and let us know how you go?

Hope this helps!
